### PR TITLE
Implement sys_getcwd()

### PR DESCRIPTION
**Pull request by dequeueing**

### CODE REVIEW

**Comment by tatetian** [comment]
Thanks for implementing this missing system call. Could you also add some gvisor system call tests for getcwd? Check out the list of system call tests [here](https://github.com/asterinas/asterinas/blob/main/test/syscall_test/Makefile).

File: kernel/src/syscall/getcwd.rs [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
 use super::SyscallReturn;
-use crate::prelude::*;
+use crate::{fs::fs_resolver::FsPath, prelude::*};
 
 pub fn sys_getcwd(buf: Vaddr, len: usize, ctx: &Context) -> Result<SyscallReturn> {
-    // TODO: getcwd only return a fake result now
-    let fake_cwd = CString::new("/")?;
-    let bytes = fake_cwd.as_bytes_with_nul();
+    let current = ctx.process;
+    let dirent = current.fs().read().lookup(&FsPath::new(-100, "").unwrap()).unwrap();
```
Comment (tatetian) [inline kernel/src/syscall/getcwd.rs:8]: Use `AT_FDCWD` instead of a magic number.

**Comment by dequeueing** [comment]
> Thanks for implementing this missing system call. Could you also add some gvisor system call tests for getcwd? Check out the list of system call tests [here](https://github.com/asterinas/asterinas/blob/main/test/syscall_test/Makefile).

If my understanding is correct, I don't think there is any test in gVisor specifically testing the function of getcwd(). But I have tested it in some application and it's working fine.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**